src/common/CMakeLists.txt

```diff
@@ -127,6 +127,7 @@ endif()
 add_library(common-common-objs OBJECT
   ${common_srcs})
 # for options.cc
+target_include_directories(common-common-objs PRIVATE ${OPENSSL_INCLUDE_DIR})
```

Comment:
So what do you need to include from openssl, that does not cause a simmilar problem on the other platforms?

Reply:
I think it's happening because we're cross compiling. If not explicitly specified, the build may try to use the system headers.

I can do a build commenting this line to point out a specific module.

Reply:
ubuntu@R01-U09-N01:~/workspace/ceph$ grep -R "include .openssl" src/common
src/common/ceph_crypto.h:#include <openssl/evp.h>
src/common/ceph_crypto.h:#include <openssl/ossl_typ.h>
src/common/ceph_crypto.h:#include <openssl/hmac.h>
src/common/ceph_crypto.cc:#include <openssl/evp.h>
src/common/ceph_crypto.cc:#  include <openssl/conf.h>
src/common/ceph_crypto.cc:#  include <openssl/engine.h>
src/common/ceph_crypto.cc:#  include <openssl/err.h>

Reply:
It probably does not hurt for Linux/BSD systems,
But you might want to either make it a WIN32-thingy or comment about this, otherwise it might disappear in a future cleanup.

Reply:
Ok, I can add a comment such as "Let's not rely on the default system headers and point Cmake to the retrieved location. This is especially important when cross compiling".

Still, explicitly stating this for every occurrence feels a bit odd. I assume one would do a git blame when questioning the reason why a certain line was added and thinking about dropping it.

Reply:
THis is really an odd one outstanding. And I'd add something like `like WIN32` in the comment.